### PR TITLE
Add a button to trigger the sync of gateway terminals

### PR DESCRIPTION
**Pull request by BushraAsif**

### CODE REVIEW

**Review by shahbaz-mehar — COMMENTED** [review]
Changes have been reviewed.

File: Controller/Adminhtml/System/Config/Button.php [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+     */
+    public function checkConfigAlreadyExist($terminalList, $scopeCode, $scopeID)
+    {
+        $i                       = 1;
+        $checkTerminalConfigured = false;
+        foreach ($terminalList as $terminal) {
+            $connection  = $this->_resource->getConnection();
+            $configExist = $connection->fetchAll('SELECT * from core_config_data WHERE path = "payment/terminal' . $i
+                                                 . '/active" AND scope = "' . $scopeCode . '" AND scope_id ='
+                                                 . $scopeID);
```
Comment (emicha) [inline Controller/Adminhtml/System/Config/Button.php:270]: This opens up for a possibility to perform SQL Injection.

Is there a way to use a parameterised request instead?

https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html

**Comment by emicha** [comment]
We should also add to the WIKI documentation that the customer can synchronise the rules by pressing such button, and show a screenshot with how to do it.

File: Controller/Adminhtml/System/Config/Button.php (hunk 1 above)
Comment (BushraAsif) [reply Controller/Adminhtml/System/Config/Button.php]: This has been updated

**Comment by BushraAsif** [comment]
> We should also add to the WIKI documentation that the customer can synchronise the rules by pressing such button, and show a screenshot with how to do it.

This has been added in the wiki.md file
https://github.com/AltaPay/plugin-magento2-community/blob/update-documentation-link/wiki.md

**Review by emicha — APPROVED** [review]

**Pull request merged**